dpnp/backend/kernels/elementwise_functions/isclose.hpp

```diff
@@ -0,0 +1,345 @@
+                    out_[k] = isclose(a_[k], b_[k], rtol_, atol_, equal_nan_);
+                }
+            }
+        }
+        else {
+            const std::uint16_t sgSize =
+                ndit.get_sub_group().get_local_range()[0];
+            const std::size_t gid = ndit.get_global_linear_id();
+            const std::uint16_t elems_per_sg = sgSize * elems_per_wi;
+
+            const std::size_t start =
+                (gid / sgSize) * (elems_per_sg - sgSize) + gid;
+            const std::size_t end = std::min(nelems_, start + elems_per_sg);
+            for (std::size_t offset = start; offset < end; offset += sgSize) {
+                out_[offset] =
+                    isclose(a_[offset], b_[offset], rtol_, atol_, equal_nan_);
+            }
+        }
+    }
+};
+
+template <typename T, typename scT>
+sycl::event
+    isclose_strided_scalar_impl(sycl::queue &exec_q,
+                                const int nd,
+                                std::size_t nelems,
+                                const dpctl::tensor::ssize_t *shape_strides,
+                                const scT rtol,
+                                const scT atol,
+                                const bool equal_nan,
+                                const char *a_cp,
+                                const dpctl::tensor::ssize_t a_offset,
+                                const char *b_cp,
+                                const dpctl::tensor::ssize_t b_offset,
+                                char *out_cp,
+                                const dpctl::tensor::ssize_t out_offset,
+                                const std::vector<sycl::event> &depends)
+{
+    dpctl::tensor::type_utils::validate_type_for_device<T>(exec_q);
+
+    const T *a_tp = reinterpret_cast<const T *>(a_cp);
+    const T *b_tp = reinterpret_cast<const T *>(b_cp);
+
+    using resTy = bool;
+    resTy *out_tp = reinterpret_cast<resTy *>(out_cp);
+
+    using IndexerT =
+        typename dpctl::tensor::offset_utils::ThreeOffsets_StridedIndexer;
+    const IndexerT indexer{nd, a_offset, b_offset, out_offset, shape_strides};
+
+    sycl::event comp_ev = exec_q.submit([&](sycl::handler &cgh) {
+        cgh.depends_on(depends);
+
+        using IsCloseFunc =
+            IsCloseStridedScalarFunctor<T, scT, resTy, IndexerT>;
+        cgh.parallel_for<IsCloseFunc>(
+            {nelems},
+            IsCloseFunc(a_tp, b_tp, out_tp, indexer, rtol, atol, equal_nan));
+    });
+    return comp_ev;
+}
+
+template <typename T,
+          typename scT,
+          std::uint8_t vec_sz = 4u,
+          std::uint8_t n_vecs = 2u>
+sycl::event
+    isclose_contig_scalar_impl(sycl::queue &exec_q,
+                               std::size_t nelems,
+                               const scT rtol,
+                               const scT atol,
+                               const bool equal_nan,
+                               const char *a_cp,
+                               ssize_t a_offset,
```

Comment:
It seems we are always passing offsets as 0. Is that intentional? If yes, then it looks we don't need it. Or we missed to pass them properly?

Reply:
Offsets are unnecessary for contig implementation,
I removed their passing and changed the function signature.
Thank you